### PR TITLE
Health state tips and UI show tips

### PR DESCRIPTION
**Pull request by MaxIsJoe**
part of #5316

### Notes:
Did some cleanup in the temp UI code and added Celsius numbers of temperatures in atmos constants.

this PR was meant to be bigger but temperature stuff took my energy away from me so ;-;

# Changelog:

CL: [New] Added 19 new protips tips for health.
CL: [New] Fixed some NRE issues with protips. They should be much more stable now, with some edge cases.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosConstants.cs [hunk 1]
```diff
@@ -25,4 +25,18 @@ public static class AtmosConstants
...
+	public const float MELTING_HEAT = 460;
+	public const float HOT_HEAT = 410;
+	public const float ABIT_HOT_HEAT = 360;
+
+	public const float CELSIUS_BARELY_COLD_HEAT = 12f;
+	public const float CELSIUS_FREEZING_HEAT = 0f;
+	public const float CELSIUS_NORMAL_HEAT = 20;
+	public const float CELSIUS_WARM_HEAT = 28f;
+	public const float CELSIUS_HOT_HEAT = 38f;
+	public const float CELSIUS_MELTING_HEAT = 60f;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosConstants.cs]: These have been incorrectly converted.

**Review by ThatDan123 — CHANGES_REQUESTED** [review]
We're discussing some changes in the discord

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosConstants.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosConstants.cs]: They're not incorrectly converted because the UI uses different numbers compared to the body (old and new health shinangins according to Bod)

Comment (corp-0) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosConstants.cs]: mind explaining a little bit more? So I get the full picture.

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosConstants.cs]: I just realized something while I was writing an explanation for this.

If the UI uses old temperature numbers, why not just update the numbers to use the ones we use already for HealthV2?
Need to investigate this more

Comment (corp-0) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosConstants.cs]: that is what I was thinking indeed

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosConstants.cs]: I misunderstood something bod said and apparently the numbers that exist already work correctly as is for now, also Dan's suggestion works, so yay.

review pls

**Comment by Bod9001** [comment]
didn't we agree that everything would be in Kelvin?

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosConstants.cs [hunk 2]
```diff
@@ -25,4 +27,18 @@ public static class AtmosConstants
...
+	public const float MELTING_HEAT = 460;
+	public const float HOT_HEAT = 410;
+	public const float ABIT_HOT_HEAT = 360;
+
+	public const float CELSIUS_BARELY_COLD_HEAT = BARELY_COLD_HEAT - Reactions.KOffsetC;
+	public const float CELSIUS_FREEZING_HEAT = FREEZING_HEAT - Reactions.KOffsetC;
+	public const float CELSIUS_NORMAL_HEAT = 20f;
+	public const float CELSIUS_WARM_HEAT = ABIT_HOT_HEAT - Reactions.KOffsetC;
+	public const float CELSIUS_HOT_HEAT = HOT_HEAT - Reactions.KOffsetC;
+	public const float CELSIUS_MELTING_HEAT = MELTING_HEAT - Reactions.KOffsetC;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosConstants.cs]: I think it'd be best to get rid of all the Celsius ones and just do:

```public const float BARELY_COLD_HEAT = -13.15 + Reactions.KOffsetC;```

If you want to see the Celsius value easily, that way all the code will expect Kelvin so we don't have it all mixed around.

File: UnityProject/Assets/Scripts/Learning/ProtipObjectTypes/ProtipObjectOnHealthStateChange.cs [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+		{
+			if(state == false || ToxinTipCooldown) return;
+			TriggerTip(ToxinEvent);
+			StartCoroutine(ToxinCooldown());
+		}
+
+		private void TriggerTemperatureTip(float state)
+		{
+			if(TempTipOnCooldown) return;
+			switch (state - Reactions.KOffsetC)
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Learning/ProtipObjectTypes/ProtipObjectOnHealthStateChange.cs]: So here just remove the ```- Reactions.KOffsetC```

and replace the constants below

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosConstants.cs (hunk 2 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosConstants.cs]: done

File: UnityProject/Assets/Scripts/HealthV2/Living/HealthStateController.cs [hunk 4]
```diff
@@ -236,6 +256,61 @@ private void SyncHealthDoll(string oldDollData, string newDollData)
...
+		private void InvokeClientTemperatureEvent(float state)
+		{
+			TemperatureEvent?.Invoke(state);
+		}
+
+		[TargetRpc]
+		private void IvokeClientPressureEvent(float state)
+		{
+			PressureEvent?.Invoke(state);
+		}
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/HealthV2/Living/HealthStateController.cs:312]: Just realised reading this through again that these would probably be better as SyncVar hooks.

If you have time to change them, otherwise i'm happy to merge it now and have it done in another PR?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/HealthV2/Living/HealthStateController.cs:312]: SyncVar hooks trigger on all clients if I remember correctly, TargetRpc only triggers on the owner.

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/HealthV2/Living/HealthStateController.cs:312]: HealthStateController is special as theres a setting on the script on the prefab so that the hooks will only trigger for the connection that owns the object

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/HealthV2/Living/HealthStateController.cs:312]: I can fix this in the next PR, right now I've opened a new branch and working on something else so i can move that there.

**Review by ThatDan123 — APPROVED** [review]
